BSP/Inc/BSP.h

```diff
@@ -0,0 +1,19 @@
+#ifndef BSP_CONF_H
+#define BSP_CONF_H
+
+typedef enum {
+    BSP_OK = 0x0,
+    BSP_ERROR = 0x1,
+    BSP_BUSY = 0x2,
+    BSP_TIMEOUT = 0x4
+} BSP_Status;
+
+// Macro for turning HAL_StatusTypeDef to our BSP error
+#define CONVERT_RETURN(x) (((x) == 0x3U) ? (BSP_TIMEOUT) : (x))
```

Comment:
will wait on merging this PR once standard config PR is merged. some of this stuff has changed